### PR TITLE
Rio de Janeiro Example

### PR DESCRIPTION
**Pull request by jamesmcclain**
## Overview

Workflow configuration for doing DeepLab-based segmentation on the Rio de Janeiro dataset.

### Checklist

- [x] Ran scripts/format_code and commited any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness
- [x] https://github.com/azavea/raster-vision/pull/366
- [x] https://github.com/azavea/raster-vision/pull/370

### CODE REVIEW

**Comment by jamesmcclain** [comment]
### Configuration ###

```
    "model_variant": "xception_65",
    "output_stride": 16,
    "atrous_rates": [6, 12, 18],
    "decoder_output_stride": 4,
    "base_learning_rate": "0.001"
```

### Ground Truth ####

label-013022223130

### Prediction ###



### Evaluation ###

```json
[
    {
        "precision": 0.7610906067895855,
        "recall": 0.628767788220516,
        "gt_count": 9042009,
        "class_name": "building",
        "f1": 0.6886302362666523,
        "class_id": 1,
        "count_error": 5141331
    },
    {
        "precision": 0.9556501173274803,
        "recall": 0.97592039021509,
        "gt_count": 74114407,
        "class_name": "~building",
        "f1": 0.9656788936976214,
        "class_id": 2,
        "count_error": 5141331
    },
    {
        "precision": 0.9344946980592997,
        "recall": 0.9381727682924671,
        "gt_count": 83156416,
        "class_name": "average",
        "f1": 0.9355540209045379,
        "class_id": null,
        "count_error": 5141331.0
    }
] 
```

**Comment by jamesmcclain** [comment]
```json
[
    {
        "precision": 0.7458118618719443,
        "recall": 0.6751462003289954,
        "f1": 0.7087065249294107,
        "count_error": 4564450.336720263,
        "gt_count": 9950901,
        "class_name": "building",
        "class_id": 1
    },
    {
        "precision": 0.9681590318960314,
        "recall": 0.9771850255101284,
        "f1": 0.9726468077807174,
        "count_error": 3811783.01616935,
        "gt_count": 100422466,
        "class_name": "~building",
        "class_id": 2
    },
    {
        "precision": 0.9481129398298188,
        "recall": 0.9499541950188037,
        "f1": 0.9488508169909705,
        "count_error": 3879641.0347866258,
        "gt_count": 110373367,
        "class_name": "average",
        "class_id": null
    }
]
```

File: src/rastervision/commands/train.py [hunk 1]
```diff
@@ -3,6 +3,139 @@
 
 class Train(Command):
     def __init__(self, ml_task, options):
+        import io
+        import base64
+
+        data = u"""IyBDb3B5cmlnaHQgMjAxOCBUaGUgVGVuc29yRmxvdyBBdXRob3JzIEFsbCBSaWdodHMgUmVzZXJ2
```
Comment (lewfish) [inline src/rastervision/commands/train.py]: What's the deal  with this data string?

File: src/rastervision/contrib/spacenet/rasterize.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+src=$3
+sink=$4
+tempfile=$(mktemp /tmp/XXXX.tif)
+
+set -x
+gdal_merge.py -createonly -init "0 255 0" -co "COMPRESS=DEFLATE" -o $tempfile $src
+gdal_rasterize -b 1 -b 2 -b 3 -burn 0 -burn 0 -burn 255 $targets $tempfile # target class
+gdal_rasterize -i -b 1 -b 2 -b 3 -burn 0 -burn 0 -burn 0 $aoi $tempfile # AOI mask
+cp $tempfile $sink
+rm $tempfile
```
Comment (lewfish) [inline src/rastervision/contrib/spacenet/rasterize.sh]: Could use some description / instructions for this script

File: src/rastervision/commands/train.py (hunk 1 above)
Comment (jamesmcclain) [reply src/rastervision/commands/train.py]: Added in this commit https://github.com/azavea/raster-vision/pull/374/commits/a2844453a0f195e2c7275c69cfffa96244017576 .  The purpose is to patch DeepLab on AWS Batch with the changes that I made elsewhere in `azavea/models` and submitted to `tensorflow/models`.  The container (or whatever it is) that runs on Batch had not picked up the changes in `azavea/models` at the time.  That may have changes.

File: src/rastervision/contrib/spacenet/rasterize.sh (hunk 2 above)
Comment (jamesmcclain) [reply src/rastervision/contrib/spacenet/rasterize.sh]: Incoming.
